src/azure-cli/azure/cli/command_modules/acr/_params.py

```diff
@@ -135,6 +152,42 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('write_enabled', help='Indicates whether write or delete operation is allowed.', arg_type=get_three_state_flag())
 
+    with self.argument_context('acr manifest') as c:
+        c.argument('registry_name', options_list=['--registry', '-r'], help='The name of the container registry. You can configure the default registry name using `az configure --defaults acr=<registry name>`', completer=get_resource_name_completion_list(REGISTRY_RESOURCE_TYPE), configured_default='acr', validator=validate_registry_name)
+        c.argument('top', type=int, help='Limit the number of items in the results.')
+        c.argument('orderby', help='Order the items in the results. Default to alphabetical order of names.', arg_type=get_enum_type(['time_asc', 'time_desc']))
+        c.argument('delete_enabled', help='Indicates whether delete operation is allowed.', arg_type=get_three_state_flag())
+        c.argument('list_enabled', help='Indicates whether this item shows in list operation results.', arg_type=get_three_state_flag())
+        c.argument('read_enabled', help='Indicates whether read operation is allowed.', arg_type=get_three_state_flag())
+        c.argument('write_enabled', help='Indicates whether write or delete operation is allowed.', arg_type=get_three_state_flag())
+        c.argument('repository', help='The name of the repository.', options_list=['--name', '-n'], validator=validate_repository)
+        c.argument('manifest_id', help="The name of the artifact. May include a tag in the format 'name:tag' or digest in the format 'name@digest'.", options_list=['--name', '-n'])
+
+    # Positional arguments must be specified on each individual command, they cannot be assigned to a command group
+    with self.argument_context('acr manifest show') as c:
+        c.positional('ID', arg_type=manifest_id_type)
+        c.argument('raw_output', help='Output the raw manifest text with no formatting.', options_list=['--raw'], action='store_true')
+
+    with self.argument_context('acr manifest list') as c:
+        c.positional('ID', arg_type=repo_id_type)
+
+    with self.argument_context('acr manifest delete') as c:
+        c.positional('ID', arg_type=manifest_id_type)
+
+    with self.argument_context('acr manifest list-referrers') as c:
+        c.positional('ID', arg_type=manifest_id_type)
+        c.argument('artifact_type', help='Filter referrers based on artifact type.')
+        c.argument('recursive', help='Recursively include referrer artifacts.', action='store_true')
+
+    with self.argument_context('acr manifest metadata show') as c:
+        c.positional('ID', arg_type=manifest_id_type)
+
+    with self.argument_context('acr manifest metadata list') as c:
+        c.positional('ID', arg_type=repo_id_type)
+
+    with self.argument_context('acr manifest metadata update') as c:
+        c.positional('ID', arg_type=manifest_id_type)
```

Comment:
Hey, I'm not sure I understand. ID is declared in the function definition of `acr_manifest_metadata_update` or do you mean I need to declare it somewhere else also? 

The function definition is on line 363 of manifest.py 

![image](https://user-images.githubusercontent.com/64281796/153444421-8b3e08a3-3577-461d-b6ed-d868d087b505.png)

I don't believe I'm using a python SDK at all here, the `acr_manifest_util` points directly at manifest.py 

![image](https://user-images.githubusercontent.com/64281796/153444820-6947f93e-43d8-4dce-94c6-5d1ef9ffe93e.png)

Reply:
Sorry, I didn't notice that before. I have seen the processing logic of this error in Linter as follows:
```Python
@ParameterRule(LinterSeverity.HIGH)
def no_positional_parameters(linter, command_name, parameter_name):
    options_list = linter.get_parameter_options(command_name, parameter_name)

    if not options_list:
        raise RuleError("CLI commands should have optional parameters instead of positional parameters. "
                        "However parameter '{}' in command '{}' is a positional."
                        .format(parameter_name, command_name))

def get_parameter_options(self, command_name, parameter_name):
    return self.get_command_metadata(command_name).arguments.get(parameter_name).type.settings.get('options_list')
```
I guess because parameter `ID` is all uppercase, so it cannot be used as the default option name, so you need to specify the option name for this parameter by `options_list=[...]`. Could you add it to have a try?

Reply:
Yes, we want the positionals to provide a consistent experience for customers who are used to using other open source tools to interact with our service such as docker, helm, kubectl etc that all use positionals in this way. Added to linter exclusions